icu4c/source/i18n/nfrule.cpp

```diff
@@ -1505,9 +1509,15 @@ NFRule::findText(const UnicodeString& str,
         return str.indexOf(key, startingAt);
     }
     else {
-        // but if lenient parsing is turned ON, we've got some work
-        // ahead of us
-        return findTextLenient(str, key, startingAt, length);
+        // Check if non-lenient rule finds the text before call lenient parsing
+        *length = key.length();
+        int pos = str.indexOf(key, startingAt);
```

Comment:
I think most of the time we try to use the exact type, like int32_t instead of the old int type. This is meant to improve portability. Though for strings, it's not that big of a deal these days.